telemetry/internal/appinsights/transmitter.go

```diff
@@ -142,28 +138,45 @@ func canRetryBackendError(berror contracts.BackendResponseError) bool {
 		berror.StatusCode == tooManyRequestsOverExtendedTimeResponse
 }
 
-func (result *transmissionResponse) getRetryItems(items []batchItem) []batchItem {
+func (result *transmissionResponse) getRetryItems(items []batchItem) (succed, failed int, retries []batchItem) {
+	defer func() {
+		for i := len(retries) - 1; i >= 0; i-- {
+			item := &retries[i]
+			// If the item has been retried too many times, we consider it failed.
+			if item.retries > 2 {
+				retries = retries[:i]
```

Comment:
This seems complicated... I'm not totally sure what `getRetryItems` is intending to do overall, and what assumptions it has. E.g. is this intended to remove item `i` and all items after it, even if some of the items after `i` had retries remaining? Or is this not a possible situation to get into?

I feel like doc comments and/or more expressive helpers are needed here.